src/rating/rating.js

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+
+import React from 'react';
+import {KIND} from './constants';
+import type {RatingPropsT, RatingStateT} from './types';
+import {StyledRoot, StyledStar, StyledEmoticon} from './styled-components';
+import {getOverrides} from '../helpers/overrides';
+
+const ENTER_KEY_CODE = 13;
+const SPACE_KEY_CODE = 32;
+
+class Rating extends React.Component<RatingPropsT, RatingStateT> {
+  static defaultProps = {
+    overrides: {},
+    kind: KIND.star,
+  };
+
+  state = {};
+
+  selectItem = (value: number) => {
+    const {onChange} = this.props;
+
+    onChange && onChange({value});
+    this.setState({previewIndex: undefined});
+  };
+
+  updatePreview = (previewIndex?: number) => {
+    this.setState({previewIndex});
+  };
+
+  renderRatingContents = () => {
+    const {overrides = {}, value = -1} = this.props;
+    const {previewIndex} = this.state;
+
+    const [Star, starProps] = getOverrides(overrides.Star, StyledStar);
+    const [Emoticon, emoticonProps] = getOverrides(
```

Comment:
Wonder if providing a single override would be more generalized and flexible here. 
Something like `overrides.RatingItem` or `overrides.RatingPoint` or something else. And then provide our default here based in the `kind` prop: `getOverrides(overrides.RatingItem, kind === KIND.star ? StyledStar : StyledEmoticon)`

Reply:
With the split to `StarRating` and `EmoticonRating` to allow dynamic sizing, the overrides now match `{ Root, Item }`